Engines stuff

comma opened the pull request:
Moves a lot of logic from engine interface datum into machinery object
Moves a lot of logic from engine control console into the ship object
can_burn() doesn't do actual burn now, so no fuel going poof.
burn() now returns actual thrust of the burn, get_thrust() estimates it.
Removed a lot of paranoid sanity from engine datum. Let us beieve in Destroy()

Changed thrust calculations. It is no longer just pressure, but sqrt(mass * pressure / 100). Different gasses give different thrust, worst being nitrogen, best being plasma.

I think that'll do for now, numbers and shit will need to be tweaked once engines are mapped in, since only hard limit is mapping space really.

mwerezak commented on code/modules/overmap/ships/engines/thermal.dm:
```diff
@@ -47,53 +42,82 @@
+	if (!is_on())
+		return 0
+	if(!check_fuel())
+		visible_message(src,"<span class='warning'>[src] coughs once and goes silent!</span>")
```
This reads like it should probably be `audible_message()`

mwerezak commented on code/modules/xgm/xgm_gas_mixture.dm:
```diff
@@ -484,3 +483,10 @@
+
+/datum/gas_mixture/proc/get_mass()
+	for(var/g in gas)
+		. += gas[g] * gas_data.molar_mass[g]
```
`. += gas[g] * gas_data.molar_mass[g] * group_multiplier` 

Just in case a gas_mixture with a different group_multiplier ever gets used.

mwerezak commented:
Looks pretty good overall, didn't see any serious problems.

comma committed:
Ovemap Engine stuff

Moves a lot of logic from engine interface datum into machinery object
Moves a lot of logic from engine control console into the ship object
can_burn() doesn't do actual burn now, so no fuel going poof.
burn() now returns actual thrust of the burn, get_thrust() estimates it.
Removed a lot of paranoid sanity from engine datum. Let us beieve in Destroy()

Changed thrust calculations. It is no longer just pressure, but sqrt(mass * pressure / 100). Different gasses give different thrust, worst being nitrogen, best being plasma.

NullSnapshot reviewed (approved):
What is plasma? :V
